examples/gnrc_networking/udp.c

```diff
@@ -106,7 +110,11 @@ static void send(char *addr_str, char *port_str, char *data, unsigned int num,
          * => use temporary variable for output */
         printf("Success: sent %u byte(s) to [%s]:%u\n", payload_size, addr_str,
                port);
+#if IS_USED(MODULE_ZTIMER_MSEC)
+        ztimer_sleep(ZTIMER_MSEC, delay / 1000);
```

Comment:
Mhhh the delay is in microseconds. I would feel much better if we could go for

```suggestion
#if IS_USED(MODULE_ZTIMER_USEC)
        ztimer_sleep(ZTIMER_USEC, delay);
```

for now, because otherwise expectations for testing or experimentation might be unfulfilled. We can change this to milliseconds in a later iteration (maybe one where we don't copy `udp.c` all over the place xD)

Reply:
(as you pointed out yourself: we can't benefit from the power consumption benefits from `ztimer_msec` anyways at the moment here).

Reply:
Thats what I had initially. I then decided for MSEC as this was the timer that is used throughout GNRC (and NimBLE) now, so it will be most likely included in the build. Now if one does not explicitly include `ztimer_usec`, the build will fail, as the example tries to link against xtimer, which is however not part of the build. By using `ztimer_msec` here we do not have that problem.

The clean way would of course be to also adapt the `udp` shell command to take its input in milliseconds, which would be feasible, but I did not want to break the behavior of the command...

Reply:
> The clean way would of course be to also adapt the `udp` shell command to take its input in milliseconds, which would be feasible, but I did not want to break the behavior of the command...

As it stands the behavior of the command is now broken as well, as with `xtimer` the user can expect the currently advertised sub-millisecond precision, while with `ztimer` they can not.

Reply:
Nobody will ever notice :-)

No seriously, you are right. This was a quick adaption which I probably haven't thought through. By now I think changing the shell command to millisecond input would be the way to go. Do you agree?

Reply:
> No seriously, you are right. This was a quick adaption which I probably haven't thought through. By now I think changing the shell command to millisecond input would be the way to go. Do you agree?

Might be confusing at first (especially since other apps do not have this scale), but from a conceptual point I more with that approach than hidding the millisecond precision.